src/api/KY/AdvertisementService/advertisement.api.ts

```diff
@@ -0,0 +1,50 @@
+import { KyApi } from '@/api/KY/meta/ApiService/ky.api';
+import {
+  ApiAdvertisementCategoryType,
+  ApiAdvertisementListItemType,
+  ApiAdvertisementType,
+  ApiPatchedAdvertisementType,
+} from '@/api/KY/AdvertisementService/advertisement.types';
+
+class AdvertisementApi extends KyApi {
+  getAllAdvertisements = async () => {
+    return this.kyCall<Array<ApiAdvertisementListItemType>>('', {
```

Comment:
Если `kyCall` возвращает промис, тогда стоит добавить `await` перед `this.kyCall`

и ещё
```suggestion
    return this.kyCall<ApiAdvertisementListItemType[]>('', {
```
Это сокращённая запись. Пробегись и проставь так везде